### ISSUE TITLE
TypeError: latLng is not a function

### ISSUE THREAD

**Issue by Williangalvani** [vue-leaflet/vue-leaflet#47]
Hi, I'm getting this issue when trying to update the position of a marker:
```
vue-leaflet.esm.js?5afd:729 Uncaught (in promise) TypeError: latLng is not a function
    at Object.setLatLng (vue-leaflet.esm.js?5afd:729)
    at eval (vue-leaflet.esm.js?5afd:95)
    at callWithErrorHandling (runtime-core.esm-bundler.js?5c40:154)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js?5c40:163)
    at Array.job (runtime-core.esm-bundler.js?5c40:2012)
    at flushPreFlushCbs (runtime-core.esm-bundler.js?5c40:306)
    at updateComponentPreRender (runtime-core.esm-bundler.js?5c40:4160)
    at componentEffect (runtime-core.esm-bundler.js?5c40:4091)
    at Object.reactiveEffect [as update] (reactivity.esm-bundler.js?a1e9:42)
    at updateComponent (runtime-core.esm-bundler.js?5c40:4006) 
```
at this line: https://github.com/vue-leaflet/vue-leaflet/blob/master/src/functions/marker.js#L63

I also get this warning when loading the page: 
`runtime-core.esm-bundler.js?5c40:38 [Vue warn]: injection "latLng" not found. `

Am I doing something wrong?

**Comment by patrickelectric** [comment]
It works with 1.2.0, but if fails with 0.2.0 and 0.3.0

**Comment by Saltibarsciai** [comment]
Same, can't update markers, but on 0.1.2 it works

**Comment by mikeu** [comment]
Thanks for these reports! Can anyone here provide a sandbox/fiddle project that reproduces this issue, or some demo code of how/when the marker is getting updated to trigger the error?

In the simple playground demo in the repo, adding a button that sets a new value on the `data` property bound to `:lat-lng` moves the marker for me, and doesn't generate any console errors or warnings (at least at 0.3.0, I didn't go back further).

I'm also not seeing that injection warning. Off the top of my head my best guess is that maybe you're trying to update the location very early in the component life cycle and the Leaflet elements haven't all finished being loaded, created, injected, etc. yet, but without a working repro it's hard to say for sure.

**Comment by Saltibarsciai** [comment]
@mikeu
I'll try to reproduce it on demo, I also tried to reproduce it on new project but it worked without issues. When updated vue-leaflet from 0.1.2 to 0.2.0/0.3.0 it stopped working on real project so I'll have to track what's wrong

Markers worked if assigned in `data() {array: someArray}` but failed once changed on @ready. 
But it's probably something else, I'll try to reproduce it on demo repo

**Comment by Saltibarsciai** [comment]
Could not reproduce yet, but found workaround, you need to remove markers before assigning new ones each time like: 
```
this.markers = [];
this.markers = response.data
```

**Comment by Saltibarsciai** [comment]
Just found that this is not good workaround, because if popup is clicked and I want to change icon I first need to delete that marker which closes popup. 

Hacks don't work very good

**Comment by Saltibarsciai** [comment]
reproduced: https://github.com/Saltibarsciai/leaflet-laravel

**Comment by Saltibarsciai** [comment]
@mikeu also found out that using fitbounds ( with a instance of Object workaround) also gives latLng is not a function. I think if figured out why that latLng error is at all on demo it will be fixed everywhere

**Comment by mikeu** [comment]
@Saltibarsciai thanks for the repo with the issue in it. I haven't figured out the solution to this issue yet, but I _have_ narrowed down what's behind it a little bit.

If I take the contents of the App.vue in the reproduction above and paste them exactly in place of Playground.vue, with the only change being to import from `"./components"` instead of from `"@vue-leaflet/vue-leaflet"`, then the code works just fine, with none of the errors or console warnings reported here.

@DonNicoJs any idea if these problems could maybe be due to something in the rollup config when building the library, or if it could be something in how the default Vue CLI webpack config treats it?

**Comment by Saltibarsciai** [comment]
@mikeu this seems like a config issue. If 0.1.2 didn't have those issues and 0.2.0 introduced them I think it's possible to trace which change breaks it. I remember that when mapObject became leafletObject, nextTicks stopped working and I had to use @ready, so maybe something is there.

I downloaded 0.1.2 and 0.2.0 and tried to somehow use phpstorms version control tools to rollback all changes one by one till it works, but couldn't setup correctly
